lib/task_listener.js

```diff
@@ -253,6 +266,7 @@ export default class TaskListener extends EventEmitter {
       this.cleanupRunningState(runningState);
       return;
     }
+    await this.recordCapacity();
```

Comment:
@EggyLv999 has a valid point that we're not calling `availableCapacity` when reporting a removed task... So we just removed what was there last time... that is not ideal...

Maybe we can add one to `lastKnownCapacity` when a task is removed...